db/version_edit.h

```diff
@@ -374,10 +376,45 @@ class VersionEdit {
     return blob_file_garbages_;
   }
 
+  // Add a WAL (either just created or closed).
+  void AddWal(WalNumber number, WalMetadata metadata = WalMetadata()) {
+    wal_additions_.emplace_back(number, std::move(metadata));
+  }
+
+  // Retrieve all the added WALs.
+  const WalAdditions& GetWalAdditions() const { return wal_additions_; }
+
+  // VersionEdits with WAL addition shouldn't contain other types of edits.
+  bool IsWalAddition() const {
+#ifndef NDEBUG
+    if (!wal_additions_.empty()) {
+      assert(NumEntries() == wal_additions_.size());
+    }
+#endif
+    return !wal_additions_.empty();
+  }
+
+  // Delete a WAL (either directly deleted or archived).
+  void DeleteWal(WalNumber number) { wal_deletions_.emplace_back(number); }
+
+  // Retrieve all the deleted WALs.
+  const WalDeletions& GetWalDeletions() const { return wal_deletions_; }
+
+  // VersionEdits with WAL deletion shouldn't contain other types of edits.
+  bool IsWalDeletion() const {
+#ifndef NDEBUG
+    if (!wal_deletions_.empty()) {
+      assert(NumEntries() == wal_deletions_.size());
+    }
+#endif
```

Comment:
This (and the one above) could be simplified to `assert(wal_deletions_.empty() || NumEntries() == wal_deletions_.size());`. Then again, I'm wondering whether we should do this in the first place, as opposed to say, making the `WalDeletion` part of the `VersionEdit` of the flush job that makes the WAL file in question obsolete. (Note: this ties back to the question above about what we want to track.)